anomaly_detector/adapters/som_storage_adapter.py

```diff
@@ -27,11 +28,14 @@ def retrieve_data(self, timespan, max_entry, false_positive):
         data, raw = self.storage.retrieve(ESStorageAttribute(timespan,
                                                              max_entry,
                                                              false_positive))
-        if len(data) == 0:
-            logging.info("There are no logs in last %s seconds", timespan)
-            return None, None
-        else:
-            return data, raw
+        while len(data) < 2: # you need minimum of two log entries
+            logging.info("There are less than 2 log entries in last %s seconds", timespan)
+            # sleep for 10 seconds
+            time.sleep(10)
+            data, raw = self.storage.retrieve(ESStorageAttribute(timespan,
```

Comment:
We already time out higher up in the code base.

https://github.com/AICoE/log-anomaly-detector/blob/f4d625a02dfa148e79165960be8b0a8e2b5e03e8/anomaly_detector/jobs/tasks.py#L65-L67

Add the time out here:
https://github.com/AICoE/log-anomaly-detector/blob/f4d625a02dfa148e79165960be8b0a8e2b5e03e8/anomaly_detector/jobs/tasks.py#L37

The bug is likely caused because in training we don't have a check in place.